web-components/svelte-react.ts

```diff
@@ -1,9 +1,30 @@
 import React, { useRef, useEffect } from "react"
-import { SvelteComponent } from "svelte"
+import type { SvelteComponent, SvelteComponentTyped } from "svelte"
 
 const eventRegex = /on([A-Z]{1,}[a-zA-Z]*)/
 const watchRegex = /watch([A-Z]{1,}[a-zA-Z]*)/
 
+/**
+ * Type Utility for inferring the prop types of a Svelte component.
+ */
+export type SvelteProps<T> = T extends SvelteComponentTyped<infer Props, any, any> ? Props : {};
+
+/**
+ * Type utility for inferring event types of a Svelte component.
+ */
+export type SvelteEvents<T> = T extends SvelteComponentTyped<any, infer Events, any> ? Events : {};
+
+/**
+ * Type utility for getting the React props of a Svelte component. This is all
+ * the Svelte props and the events mapped to the React `onEventName` format.
+ *
+ * For example, the event `click` would be mapped to `onClick`.
+ */
+export type SvelteToReactProps<SvelteComponent> = SvelteProps<SvelteComponent> & {
+  [Prop in keyof SvelteEvents<SvelteComponent> as `on${Capitalize<Prop & string>}`]?: SvelteEvents<SvelteComponent>[Prop];
+}
```

Comment:
This is almost exactly what I was trying to achieve, but it failed short because whilst it provided code-time intellisense, it wouldn't provide build errors during storybook / webpack / tsc. Can you confirm this works in those scenarios?

Reply:
I'll have a play and confirm.